resources/views/pages/ticket/[ticket].blade.php

```diff
@@ -0,0 +1,423 @@
+                                    if ($comment['User'] != 'Server') {
+                                        $lastComment = $comment;
+                                    }
+                                }
+                            @endphp
+                            @if ($lastComment != null && ($lastComment['User'] === $user->User || $lastComment['User'] === $ticket->achievement->Author))
+                                <option value="{{ TicketAction::Request }}">Transfer to reporter - {{ $ticket->reporter->User }}</option>
+                            @endif
+                        @else
+                            <option value="{{ TicketAction::Reopen }}">Transfer to author - {{ $ticket->achievement->Author }}</option>
+                        @endif
+                        <option value="{{ TicketAction::Demoted }}">Demote achievement to Unofficial</option>
+                        <option value="{{ TicketAction::Network }}">Close - Network problems</option>
+                        <option value="{{ TicketAction::NotEnoughInfo }}">Close - Not enough information</option>
+                        <option value="{{ TicketAction::WrongRom }}">Close - Wrong ROM</option>
+                        <option value="{{ TicketAction::UnableToReproduce }}">Close - Unable to reproduce</option>
+                        <option value="{{ TicketAction::ClosedMistaken }}">Close - Mistaken report</option>
+                        <option value="{{ TicketAction::ClosedOther }}">Close - Another reason (add comments above)</option>
+                    </select>
+                    <button class='btn'>Perform action</button>           
+                @else
+                    <input type="hidden" name="action" value="{{ TicketAction::Reopen }}">
+                    <button class='btn'>Reopen this ticket</button>
+                @endif
+            @elseif ($user?->id === $ticket->reporter->id)
+                @if (TicketState::isOpen($ticket->ReportState))
+                    <input type="hidden" name="action" value="{{ TicketAction::ClosedMistaken }}">
+                    <button class='btn'>Close as mistaken report</button>
+                @endif
+            @endif
+        </form>
+
+        @if ($permissions >= Permissions::JuniorDeveloper)
+            <div class="mt-4 w-full relative flex gap-x-3">
+                <button id="achievementLogicButton" class="btn"
+                        onclick="toggleExpander('achievementLogicButton', 'achievementLogicContent')">Achievement Logic ▼</button>
+            </div>
+
+            <div id="achievementLogicContent" class="hidden devboxcontainer">
+                <div style='clear:both;'></div>
+                <li>Achievement ID: {{ $ticket->achievement->id }}</li>
+                <div>
+                    <li>Mem:</li>
+                    <code>{{ $ticket->achievement->MemAddr }}</code>
+                    <li>Mem explained:</li>
```

Comment:
I know this pattern is used elsewhere, but this HTML is technically invalid.

An `<li>` cannot exist outside of a `<ul>` or `<ol>`. It's a list item that doesn't belong to a list.

I assume in the legacy codebase, an `<li>` was chosen purely for the sake of adding a dot element next to the text. I restructured the HTML a bit here to make it valid:

```php
<div id="achievementLogicContent" class="hidden devboxcontainer">
    <ul class="list-disc ml-4 mb-2">
        <li>Achievement ID: {{ $ticket->achievement->id }}</li>
        <li>
            Mem:
            <code>{{ $ticket->achievement->MemAddr }}</code>
        </li>
    </ul>

    <p>Mem explained:</p>
    <div>
        @php
            $triggerDecoderService = new TriggerDecoderService();
            $groups = $triggerDecoderService->decode($ticket->achievement->MemAddr);
            $triggerDecoderService->addCodeNotes($groups, $ticket->achievement->GameID);
        @endphp
        <x-trigger.viewer :groups="$groups" />
    </div>
</div>
```

Note: this does cause longer code samples to immediately break to the next line.

Reply:
done